### PR TITLE
[kubernetes] add prometheus receiver docs

### PR DESCRIPTION
**Pull request by TylerHelmuth**
Adds documentation for the prometheus receiver.

Related to https://github.com/open-telemetry/opentelemetry.io/issues/3005

### CODE REVIEW

File: content/en/docs/kubernetes/collector/components.md
Comment (dashpole) [inline content/en/docs/kubernetes/collector/components.md:669]: By kubernetes metrics, do you mean metrics from kubernetes components (e.g. kubelet), or metrics from applications running on k8s?  If the former, https://kubernetes.io/docs/reference/instrumentation/metrics/ may be helpful to link to.

File: content/en/docs/kubernetes/collector/components.md
Comment (dashpole) [inline content/en/docs/kubernetes/collector/components.md:684]:
```suggestion
[Prometheus Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusreceiver).
```
typo?

File: content/en/docs/kubernetes/collector/components.md
Comment (TylerHelmuth) [reply content/en/docs/kubernetes/collector/components.md:669]: I'm not a Prometheus user myself, but my understanding was that both sources of metrics could be collected as Prometheus metrics.  Is that accurate?

File: content/en/docs/kubernetes/collector/components.md
Comment (TylerHelmuth) [reply content/en/docs/kubernetes/collector/components.md:684]: ya

File: content/en/docs/kubernetes/collector/components.md
Comment (dashpole) [reply content/en/docs/kubernetes/collector/components.md:669]: Yes, that is accurate.

**Review by dashpole — APPROVED** [review]

File: content/en/docs/kubernetes/collector/components.md
Comment (chalin) [inline content/en/docs/kubernetes/collector/components.md:682]:
```suggestion
- `alert_config.alertmanagers`
- `alert_config.relabel_configs`
- `remote_read`
- `remote_write`
- `rule_files`
```

File: content/en/docs/kubernetes/collector/components.md
Comment (TylerHelmuth) [reply content/en/docs/kubernetes/collector/components.md:669]: I'll call them both out specifically

File: content/en/docs/kubernetes/collector/components.md
Comment (chalin) [inline content/en/docs/kubernetes/collector/components.md]:
```suggestion
It supports the full set of Prometheus `scrape_config` options.
```

Or maybe link to https://prometheus.io/docs/prometheus/1.8/configuration/configuration/#scrape_config?

File: content/en/docs/kubernetes/collector/components.md
Comment (chalin) [inline content/en/docs/kubernetes/collector/components.md:25]: Copy-paste error:
```suggestion
- [Prometheus Receiver](#prometheus-receiver): receives metrics in
```

In know that we wouldn't get the same effect, but the page has an autogenerated TOC:

> Screen Shot 2023-07-31 at 17 04 35

Might it be worth dropping this autogenerated one?

Actually, if ever this page is split into separate pages, then we'll get the equivalent of the manually added TOC above automatically for the landing page -- i.e., including the description.

File: content/en/docs/kubernetes/collector/components.md
Comment (chalin) [inline content/en/docs/kubernetes/collector/components.md:672]:
```suggestion
Prometheus `scrape_config` options.
```

Or link to https://prometheus.io/docs/prometheus/1.8/configuration/configuration/#scrape_config?

File: content/en/docs/kubernetes/collector/components.md
Comment (TylerHelmuth) [reply content/en/docs/kubernetes/collector/components.md]: Linked

File: content/en/docs/kubernetes/collector/components.md
Comment (chalin) [inline content/en/docs/kubernetes/collector/components.md:5]: @TylerHelmuth - please re-run `npm run fix:dict` and commit changes.

- The PR will fail otherwise once the following lands: #3116

**Review by chalin — APPROVED** [review]
Other than the need to run `fix:dict`, LGTM

File: content/en/docs/kubernetes/collector/components.md
Comment (cartermp) [inline content/en/docs/kubernetes/collector/components.md:679]:
```suggestion
Prometheus is a common metrics format for both Kubernetes and services running
on Kubernetes. The Prometheus receiver is a minimal drop-in
replacement for the collection of those metrics. It supports the full set of
Prometheus
[`scrape_config` options](https://prometheus.io/docs/prometheus/1.8/configuration/configuration/#scrape_config).
```

File: content/en/docs/kubernetes/collector/components.md
Comment (cartermp) [inline content/en/docs/kubernetes/collector/components.md:682]:
```suggestion
The receiver returns an error if the configuration YAML/code contains any of
```

File: content/en/docs/kubernetes/collector/components.md
Comment (cartermp) [inline content/en/docs/kubernetes/collector/components.md:703]:
```suggestion
- The collector cannot auto-scale the scraping process when multiple replicas
   of the collector are run.
- When running multiple replicas of the collector with the same config, it will
  scrape the targets multiple times.
- Users need to configure each replica with a different scraping configuration if
  they want to manually shard the scraping process.
```

File: content/en/docs/kubernetes/collector/components.md
Comment (cartermp) [inline content/en/docs/kubernetes/collector/components.md:708]:
```suggestion
To make configuring the Prometheus receiver easier, the OpenTelemetry Operator
includes an optional component called the
[Target Allocator](../../operator/target-allocator). This component can be used
to tell a collector which Prometheus endpoints it should scrape.
```

**Review by cartermp — APPROVED** [review]
Overall looks good -- some quick copy edits

**Pull request merged**
